### DISCUSSION TITLE
I fixed Drop.json but don't know how to PR...

### DISCUSSION THREAD

**Discussion by FionViber** [Melledy/Grasscutter#1801] [General]
I fixed tons of errors to make the drop system to accord with that in the actual game.

But I don't know how to PR, as Drop.json is too big to edit, so I just send my commit here.

[Drop.txt](https://github.com/Grasscutters/Grasscutter/files/9633236/Drop.txt)

Here's the description:

Generally, all monsters will not drop the following items anymore, though I don't understand why they will initially: 

221 : Masterless Starglitter
222 : Masterless Stardust
201 : Primogem
305 : Anemo Sigil

Monsters which cost 体力 to get reward will not drop items anymore. Those monsters include world bosses and weekly bosses.

The following monsters will not drop items anymore, as they would not in the actual game (To shorten the list, only one ID will be shown if there are multiple IDs for a same monsters):

25090201:Maguu Kenki:Lone Gale
25090301:Maguu Kenki:Galloping Frost
25090401:Maguu Kenki:Mask of Terror
25050101:Millelith Soldier
25050201:Millelith Sergeant
26080101:Beisht
20011601:Hydro Hypostasis Summon:Split (L)
20011701:Hydro Hypostasis Summon:Split (S)
20011801:Hydro Hypostasis Summon:Self-Destruct
20011901:Hydro Hypostasis Summon:Heal
20050201:Oceanid Boar
20050301:Oceanid Crane
20050401:Oceanid Crab
20050501:Oceanid Finch
20050601:Oceanid Mallard
20050701:Oceanid Ferret
20050801:Oceanid Frog
20050901:Oceanid Raptor
25080401:Yoriki Samurai
25010103:Treasure Hoarders:Liuliu (and other Treasure Hoarders appeared in plots only)
25050101:Millelith Soldier
25050201:Millelith Sergeant
25050301:Shogunate Infantry
25050401:Shogunate Infantry Captain
25050501:Sangonomiya Cohort

The following monsters now will drop correct items:

All Samachurls
All Nobushis and Kairagis
26060101:Electro Cicin
26060201:Hydro Cicin
26060301:Cryo Cicin
21011501:Unusual Hilichurl

Errors awaiting further corrections, which are difficult to fixed without tools (yes, I fixed the above errors by hands):

Some Churls will drop wrong items. There are too many Churls, and it takes too much time to check everyone of them without a tool.

Monsters added after v2.8 are not in the list. There are too many new monsters, and it takes too much time to add them without a tool.

Some Bathysmal Vishaps are not in the list. It takes too much time to test and add them without a tool.

Some Fatuis will drop wrong items. There are too many Fatuis, and it takes too much time to check everyone of them without a tool.

**Comment by jie65535** [comment]
You can make changes to the file through the online editor and commit, and then you can submit a PR.

https://github.dev/Grasscutters/Grasscutter/blob/development/src/main/resources/defaults/data/Drop.json

_I plan to add a drop editor to my command generator, not so fast since I'm refactoring it in WPF._

**Reply by FionViber** [reply]
Wow, I know you are the developer of GC Tool, thank you for the help!

For a drop editor, now the problem is, there are multiple IDs for the same monster, I think it's because some monsters in plots or quests will not drop anything, while they will in the bigworld. So I suggest that, your drop editor need to separate regular monsters from all monsters.

Here's my idea: suppose there are 4 Monster IDs for a Hilichurl, your editor needs to tell users which ID is for the Hilichurl in the BigWorld, and suggest them to not add anything to the drop list of other 3 IDs.

**Reply by jie65535** [reply]
I opened an [issue](https://github.com/jie65535/GrasscutterCommandGenerator/issues/121).
